stub/src/main/java/io/grpc/kotlin/Helpers.kt

```diff
@@ -75,3 +77,17 @@ internal fun <T> Flow<T>.singleOrStatusFlow(expected: String, descriptor: Any):
  */
 internal suspend fun <T> Flow<T>.singleOrStatus(expected: String, descriptor: Any): T =
   singleOrStatusFlow(expected, descriptor).single()
+
+/**
+ * Returns gRPC Metadata. Throws [StatusException] if [MetadataCoroutineContextInterceptor] is not
+ * injected to gRPC server.
+ */
+suspend fun grpcMetadata(): Metadata {
+  val metadataElement =
+    coroutineContext[MetadataElement]
+      ?: throw Status.INTERNAL.withDescription(
```

Comment:
I don't think we should use INTERNAL for this because this error is caused by the user not providing the metadata.

How about  INVALID_ARGUMENT or FAILED_PRECONDITION or something else?

Reply:
AH ok, the interceptor must be provided by the server dev, that is reasonable.